Add 359 release notes (#8420)
trinodb/trino: docs/src/main/sphinx/release/release-359.md

> # Release 359 (XX Jul 2021)

Reviewer: Jun ?

## General

* Add support for row pattern recognition in window specification. ({issue}`8141`)
* Add support for {doc}`/sql/set-time-zone`. ({issue}`8112`)
* Add {func}`geometry_nearest_points`. ({issue}`8280`)
* Add support for `varchar`, `varbinary` and `date` types to {func}`make_set_digest`. ({issue}`8295`)
* Add support for granting `UPDATE` privileges. ({issue}`8279`)

> * List materialized view columns in `information_schema.columns` table ({issue}`8113`)

Reviewer 2: ```suggestion
* List materialized view columns in `information_schema.columns`. table ({issue}`8113`)
```

* Return view's and materialized view's comments in `system.metadata.table_comments`. Previously, the 
  views and materialized views were reported as if having no comment. ({issue}`8327`)
* Reduce query memory footprint. ({issue}`8086`)
* Fix query failure for certain queries with `ORDER BY ... LIMIT` on sorted data. ({issue}`8184`)

> * Fix incorrect query results for certain queries with `WHERE` using `LIKE` with pattern against 
>   `char` columns. ({issue}`8311`)

Reviewer 2: Maybe reword to? The current one is a bit of a tongue twister.

```suggestion
* Fix incorrect query results for certain queries using `LIKE` with pattern against 
  `char` columns as predicate in the `WHERE` clause. ({issue}`8311`)
```

* Fix planning failure when using {func}`hash_counts`. ({issue}`8248`)
* Fix error message when grouping expressions in ``GROUP BY`` queries contain aggregations, window functions or grouping operations. ({issue}`8247`)

## Security

> * Fix user mapping for password authentication. Formerly, user mapping for password authentication 

Author: I would like the sentence to describe what was fixed, though. Without the second part, it's too vague. Maybe something like:

> Fix spurious impersonation check when applying user mapping for password authentication.

  would result in a spurious impersonation check. ({issue}`7027`)

> * Fix handling of multiple LDAP user bind patterns. ({issue}`8134`)

Reviewer 2: If needed we can expand similar to previous bullet? WDYT?

```suggestion
* Fix handling of multiple LDAP user bind patterns. Formerly, if the first bind pattern causes an authentication failure the next pattern wasn't tried. ({issue}`8134`)
```

## Web UI

* Show session timezone in query details page. ({issue}`4196`)

## BigQuery connector

* Add `bigquery.views-cache-ttl` config property to allow configuring the cache expiration for BigQuery views. ({issue}`8236`)
* Fix incorrect results when accessing BigQuery records with wrong index. ({issue}`8183`)

## Elasticsearch connector

* Fix potential incorrect results when queries contain an `IS NULL` predicate. ({issue}`3605`)
* Fix failure when multiple indexes share the same alias. ({issue}`8158`)

## Hive connector

* Add support for Hive views which use `GROUP BY` over a subquery that also uses `GROUP BY` on matching columns. ({issue}`7635`)
* Add support for granting `UPDATE` privileges when `hive.security=sql-standard` is used. ({issue}`8279`)
* Fix query failure when reading from a non-ORC insert-only transactional table. ({issue}`8259`)
* Correct how delete markers are interpreted for ORC ACID tables,
  so Trino is not filtering out extraneous rows for certain tables. ({issue}`8208`)
* Improve performance when reading partition information from Glue metastore. ({issue}`8320`)

> * Respect `hive.metastore.glue.get-partition-threads` configuration property. ({issue}`8320`)  

Reviewer 2: These two are actually the same issue. Maybe merge the bullets? Since this was technically a regression I'm in favour of keeping 2nd bullet and dropping the first.

  
## Iceberg connector

* Do not include Hive views in `SHOW TABLES` query results. ({issue}`8153`)

## MongoDB connector

> * Skip creating an index for `_schema` collection if it already exists. Previously, it caused an 
>   unauthorized error when an user doesn't have the enough privilege ({issue}`8264`)

Reviewer 2: Maybe reword?

```suggestion
* Skip creating an index for `_schema` collection if it already exists. Previously, it caused an 
  error when the Trino user didn't have the enough privileges. ({issue}`8264`)
```

# Release 359 (XX Jul 2021)

## General

* Add support for row pattern recognition in window specification. ({issue}`8141`)
* Add support for {doc}`/sql/set-time-zone`. ({issue}`8112`)
* Add {func}`geometry_nearest_points`. ({issue}`8280`)
* Add support for `varchar`, `varbinary` and `date` types to {func}`make_set_digest`. ({issue}`8295`)
* Add support for granting `UPDATE` privileges. ({issue}`8279`)

> * Raise minimum required Java version for running Trino server to 11.0.11. ({issue}`8103`)

Reviewer 2: Maybe move this to the top of this section?

* Add support for inserting data into CSV and TEXT tables with `skip_header_line_count` table property set to 1.
  The same applies to creating tables with data using `CREATE TABLE ... AS SELECT` syntax. ({issue}`8390`)
* Disallow creating CSV and TEXT tables with data if `skip_header_line_count` is set to a value
  greater than 0. ({issue}`8373`)
* Fix query failure when reading from a non-ORC insert-only transactional table. ({issue}`8259`)
* Correct how delete markers are interpreted for ORC ACID tables,
  so Trino is not filtering out extraneous rows for certain tables. ({issue}`8208`)
* Respect `hive.metastore.glue.get-partition-threads` configuration property. ({issue}`8320`)

## Iceberg connector

* Do not include Hive views in `SHOW TABLES` query results. ({issue}`8153`)

## MongoDB connector

* Skip creating an index for the `_schema` collection if it already exists. Previously, it caused an
  error when the Trino user did not have enough privileges. ({issue}`8264`)

## MySQL connector

* Support reading and writing `timestamp` values with precision higher than 3. ({issue}`6910`)
* Support predicate pushdown on `timestamp` columns. ({issue}`7413`)

> * Handle `timestamp` values during forward offset changes ('gaps' in DST). ({issue}`5449`)

Reviewer 2: Sounds incomplete (I know I added this note myself 🙂).
```suggestion
* Handle `timestamp` values during forward offset changes ('gaps' in DST) correctly. ({issue}`5449`)
```

# Release 359 (1 Jul 2021)

## General

* Raise minimum required Java version for running Trino server to 11.0.11. ({issue}`8103`)
* Add support for row pattern recognition in window specification. ({issue}`8141`)
* Add support for {doc}`/sql/set-time-zone`. ({issue}`8112`)
* Add {func}`geometry_nearest_points`. ({issue}`8280`)
* Add {func}`current_groups`. ({issue}`8446`)  
* Add support for `varchar`, `varbinary` and `date` types to {func}`make_set_digest`. ({issue}`8295`)
* Add support for granting `UPDATE` privileges. ({issue}`8279`)
* List materialized view columns in the `information_schema.columns` table. ({issue}`8113`)
* Expose comments in views and materialized views in `system.metadata.table_comments` correctly. ({issue}`8327`)

> * Reduce query memory footprint. ({issue}`8086`)

Author: I agree. I'll remove it for now. We can add it later if we can come up with a more precise description.

* Fix incorrect results when accessing BigQuery records with wrong index. ({issue}`8183`)

## Elasticsearch connector

* Fix potential incorrect results when queries contain an `IS NULL` predicate. ({issue}`3605`)
* Fix failure when multiple indexes share the same alias. ({issue}`8158`)

## Hive connector
                         
* Rename `hive-hadoop2` connector to `hive`. ({issue}`8166`)
* Add support for Hive views which use `GROUP BY` over a subquery that also uses `GROUP BY` on matching columns. ({issue}`7635`)
* Add support for granting `UPDATE` privileges when `hive.security=sql-standard` is used. ({issue}`8279`)
* Add support for inserting data into CSV and TEXT tables with `skip_header_line_count` table property set to 1.
  The same applies to creating tables with data using `CREATE TABLE ... AS SELECT` syntax. ({issue}`8390`)
* Disallow creating CSV and TEXT tables with data if `skip_header_line_count` is set to a value
  greater than 0. ({issue}`8373`)
* Fix query failure when reading from a non-ORC insert-only transactional table. ({issue}`8259`)
* Correct how delete markers are interpreted for ORC ACID tables,

>   so Trino is not filtering out extraneous rows for certain tables. ({issue}`8208`)

Reviewer 3: How about
```
Fix incorrect results when reading ORC ACID tables containing deleted rows.
```

* Respect `hive.metastore.glue.get-partition-threads` configuration property. ({issue}`8320`)

## Iceberg connector

* Do not include Hive views in `SHOW TABLES` query results. ({issue}`8153`)

## MongoDB connector

* Skip creating an index for the `_schema` collection if it already exists. Previously, it caused an

>   error when the Trino user did not have enough privileges. ({issue}`8264`)

Reviewer 3: drop the second sentence
